### ISSUE TITLE
Color blending does not work with pdf

### ISSUE THREAD

**Issue by btracey** [gonum/plot#421]
Sample code (from @sbinet ):

```
package main

import (
	"flag"
	"fmt"
	"image/color"
	"log"
	"math"
	"math/rand"

	"gonum.org/v1/plot"
	"gonum.org/v1/plot/plotter"
	"gonum.org/v1/plot/vg"
)

func main() {

	vv := flag.Int("alpha", 128, "alpha value for transparency")
	flag.Parse()
	alpha := uint8(*vv)

	n := 10000
	vals1 := make(plotter.Values, n)
	vals2 := make(plotter.Values, n)
	for i := 0; i < n; i++ {
		vals1[i] = rand.NormFloat64()
		vals2[i] = rand.NormFloat64() + 1
	}

	p, err := plot.New()
	if err != nil {
		log.Panic(err)
	}
	p.Title.Text = fmt.Sprintf("Histogram -- alpha=%d", alpha)

	h1, err := plotter.NewHist(vals1, 16)
	if err != nil {
		log.Panic(err)
	}
	h1.LineStyle.Color = color.Black
	h1.FillColor = color.NRGBA{255, 0, 0, alpha}
	h1.Normalize(1)
	h2, err := plotter.NewHist(vals2, 16)
	if err != nil {
		log.Panic(err)
	}
	h2.LineStyle.Color = color.Black
	h2.FillColor = color.NRGBA{0, 0, 255, alpha}
	fmt.Println("alpha = ", alpha)
	h2.Normalize(1)

	p.Add(h1, h2)

	p.BackgroundColor = color.Opaque

	const (
		height = 10 * vg.Centimeter
		width  = height * math.Phi
	)

	err = p.Save(width, height, "plot.png")
	if err != nil {
		log.Fatal(err)
	}
}
```

When you save as a png, you get nicely blended histogram values

![plot](https://user-images.githubusercontent.com/3680859/33728850-0f880998-db39-11e7-98fa-7abe22e733ca.png)

When you save as a PDF, the colors blend toward black instead of toward white, and the colors do not blend together at all, just overwrite one another

[plot.pdf](https://github.com/gonum/plot/files/1540000/plot.pdf)

**Comment by sbinet** [comment]
FYI, I have implemented opacity in the new (still WIP) PDF backend that uses `github.com/jung-kurt/fpdf`:
https://github.com/sbinet-gonum/plot/commit/0c722f19a3356a7c54e019e63e73fe38b7188cee

as I said on slack, `zombiezen/pdf` doesn't seem to provide an API to change the opacity.

**Comment by btracey** [comment]
@sbinet This is fixed by #433 , agreed?

**Comment by sbinet** [comment]
indeed:

- [old-plot.pdf](https://github.com/gonum/plot/files/2001355/old-plot.pdf)
- [new-plot.pdf](https://github.com/gonum/plot/files/2001356/new-plot.pdf)
